### DISCUSSION TITLE
Remote development / collab best practice question

### DISCUSSION THREAD

**Discussion by nickcoppins** [statamic/cms#8405] [General]
Hi there, 

We are starting to develop our first company Statamic site and loving it. 

I'm working with a remote developer and just have a question about best practice. 

- We have a github repo setup and a site deployed via ploi.io. 
- Deploy script working, Statamic site up and running. 
- Statamic/Git working OK and auto-committing & pushing on save back to the github repo. 

We run into an issue where someone makes changes on the deployed server (editing collections, adding blueprints etc) and the dev has made changes on their local and pushed them to the repo whereafter the server / Statamic site won't be able to commit it's changes without first fetching and it seems to get stuck with errors like "Updates rejected because tip of branch is behind it's remote counterpart" etc. 

Is there a best practice / suggestion for this kind of mixed development (local and on-server) or are we just asking for trouble? Or am I totally missing something? 

Thank you in advanced.

**Comment by jackmcdade** [comment]
Hey there, glad you're enjoying it!

The best practice here would be for your devs to always `git pull --rebase` before pushing their commits. That'll make sure everything is synced up properly.

**Reply by nickcoppins** [reply]
Great thank you Jack.
